app/jobs/webhooks/whatsapp_events_job.rb

```diff
@@ -2,12 +2,22 @@ class Webhooks::WhatsappEventsJob < ApplicationJob
   queue_as :default
 
   def perform(params = {})
-    return unless params[:phone_number]
+    channel = find_channel(params)
+    return if channel.blank?
+
+    case channel.provider
+    when 'whatsapp_cloud'
+      Whatsapp::IncomingMessageWhatsappCloudService.new(inbox: channel.inbox, params: params).perform
+    else
+      Whatsapp::IncomingMessageService.new(inbox: channel.inbox, params: params).perform
```

Comment:
The default service would be `360dialogu`, that's being handled here but is this change affecting WhatsApp integration through Twilio?

Reply:
Twilio WhatsApp works under a separate channel type at the moment.. `Channel::Twilio` .
since provider `default` is connected to 360 dialog, went with this. lets refactor this one when we look into other providers